ndmg/scripts/ndmg_dwi_pipeline.py

```diff
@@ -410,6 +414,53 @@ def ndmg_dwi_worker(
                     "s",
                 )
             )
+        
+        #Check that the atlas hasn't lost any of the rois
+        if reg_style == "native":
+            labels_im_file_dwi_list = []
+            for idx, label in enumerate(labels):
+                labels_im_file = gen_utils.reorient_img(labels[idx], namer)
+                labels_im_file = gen_utils.match_target_vox_res(
+                    labels_im_file, vox_size, namer, sens="t1w"
+                )
+                orig_lab = nib.load(labels_im_file)
+                orig_lab = orig_lab.get_data().astype("int")
+                n_ids = orig_lab[orig_lab>0]
+                num = len(np.unique(n_ids))
+
+                labels_im_file_dwi = reg.atlas2t1w2dwi_align(labels_im_file, dsn=False)
+                labels_im = nib.load(labels_im_file_dwi)
+                align_lab = labels_im.get_data().astype("int")
+                n_ids_2 = align_lab[align_lab>0]
+                num2 = len(np.unique(n_ids_2))
+
+                if num != num2:
+                    raise KeyError('The atlas has lost an roi due to alignment')
+
+                labels_im_file_dwi_list.append(labels_im_file_dwi)
+        elif reg_style == "native_dsn":
+            labels_im_file_mni_list = []
+            for idx, label in enumerate(labels):
+                labels_im_file = gen_utils.reorient_img(labels[idx], namer)
+                labels_im_file = gen_utils.match_target_vox_res(
+                    labels_im_file, vox_size, namer, sens="t1w"
+                )
+                orig_lab = nib.load(labels_im_file)
+                orig_lab = orig_lab.get_data().astype("int")
+                n_ids = orig_lab[orig_lab>0]
+                num = len(np.unique(n_ids))
+
+                labels_im_file_mni = reg.atlas2t1w2dwi_align(labels_im_file, dsn=True)
+                labels_im = nib.load(labels_im_file_mni)
+                align_lab = labels_im.get_data().astype("int")
+                n_ids_2 = align_lab[align_lab>0]
+                num2 = len(np.unique(n_ids_2))
+
+                if num != num2:
+                    raise KeyError('The atlas has lost an roi due to alignment')
+
+                labels_im_file_mni_list.append(labels_im_file_mni)
```

Comment:
Is it possible to just stick all this stuff into a function that you import from somewhere, and then call with slightly different arguments for `native` and `native_dsn`? in general repeating code is something we should try to avoid